google/cloud/bigquery/_helpers.py

```diff
@@ -82,7 +82,18 @@ def _timestamp_from_json(value, field):
     """Coerce 'value' to a datetime, if set or not nullable."""
     if _not_null(value, field):
         # value will be a float in seconds, to microsecond precision, in UTC.
-        return _datetime_from_microseconds(1e6 * float(value))
+
+        if value is not None:
+            if type(value) == int:
+                value = value
+            elif type(value) == float:
+                value = 1e6 * float(value)
+            elif value.isdigit():
+                value = int(value)
+            else:
+                value = 1e6 * float(value)
```

Comment:
Why is this case needed? Couldn't the above `isdigit()` check be removed and we always fall back to converting to integer? I don't expect floating point values to be serialized as strings in JSON.